sdk/rust/src/http/router.rs

```diff
@@ -321,6 +333,15 @@ mod tests {
         assert_eq!(res.into_body().unwrap(), "foo".to_string());
     }
 
+    #[test]
+    fn test_router_display() {
+        let mut router = Router::default();
+        router.get("/:x", echo_param);
+        
+        let expected = "Registered routes:\n- GET: /:x\n";
+        let actual = format!("{}", router);
+        assert_eq!(actual, expected.to_string());
```

Comment:
Nit:
```suggestion
        let expected = "Registered routes:\n- GET: /:x\n";
        let actual = format!("{}", router);
        assert_eq!(actual.as_str(), expected);
```

Reply:
@rylev you comment got my attention 😄 

I guess your mentioning this because (IMO) `expected.to_string()` is more expensive. I assume `.to_string()` doing additional heap allocation (due to calling `String::from("")` internally)

The implementation of `as_str()` on `String` simply returns `self`, which I interpret as a borrowed `&str` with "potentially" shorter lifetime.  

So the change would prevent an additional heap allocation. Is my interpretation here correct, or did I get something wrong?

Reply:
Everything you said is more or less correct. However I made the suggestion just because it's shorter 😀, and I find it easier to read. After all, one heap allocation in a single test won't really make any difference at all in the performance of the test suite.

Reply:
Thanks. Just wanted to understand it. shorter is better! agree 💯 on that 😄 Was just curious.